android-core/src/main/java/com/mparticle/internal/ConfigManager.java

```diff
@@ -1213,7 +1198,7 @@ public void setIdentityApiContext(String context) {
     public String getPreviousAdId() {
         MPUtility.AdIdInfo adInfo = MPUtility.getAdIdInfo(mContext);
         String currentAdId = null;
-        if (adInfo != null) {
+        if (adInfo != null && !adInfo.isLimitAdTrackingEnabled) {
```

Comment:
so reading this code - this is what could happen:
1. user has limit ad tracking (LAT) false, and we persist the Ad ID (via `setPreviousAdId` below)
2. they then disable limit ad tracking
3. this method is called - and we skip over this block with your change - so it will return the previous Ad ID, which we should no longer be using 

So that makes me nervous. Do we need these methods at all? Why are we storing the ad ID? If these methods are used to understand if the Ad ID changed, then I would think you'd want this method to return all zeroes or null when LAT is true, so that whatever caller knows that the Ad ID changed from something to null.

Are we triggering a modify call to identity based on these methods? If we are, i'd expect that modify call to remove the Ad ID from the user.

Reply:
@samdozor in `AppStateManager.java` updated the logic to allow it to send at least one modify call if currentGoogleAdId is `null`